### PR TITLE
Uses new XML:DB api binary as available on maven central

### PR DESCRIPTION
**Pull request by reinhapa**
### Description:
Use of the official XML:DB API binary including the changes done by the eXist-db team with the following additional enhancements:

- Use of `java.utility.concurrent.ConcurrentHashMap` instead of `java.util.Hashtable` within the `org.xmldb.api.DatabaseManager` to hold the registered `org.xml.api.base.Database` instances.
- Adds ` org.xmldb.api.base.Collection` additionally extends `java.lang.AutoClosable` to support try-with-resources.

### CODE REVIEW

**Comment by duncdrum** [comment]
tests are failing

File: extensions/commands/src/org/exist/commands/info/Lock.java [hunk 1]
```diff
@@ -53,7 +53,8 @@ public void process(XmldbURI collectionURI, String[] commandData) throws Command
 				Collection collection = broker.getCollection(collectionURI);
 
 				out().println("Collection lock:");
-				collection.getLock().debug(out());
+				//TODO:check where that method is
+				//collection.getLock().debug(out());
```
Comment (adamretter) [inline extensions/commands/src/org/exist/commands/info/Lock.java]: What is this change for? Is this still outstanding?

File: extensions/commands/src/org/exist/commands/info/Lock.java [hunk 2]
```diff
@@ -68,7 +69,8 @@ public void process(XmldbURI collectionURI, String[] commandData) throws Command
 				out().println("Lock token: " + doc.getMetadata().getLockToken());
 
 				out().println("Update lock: ");
-				doc.getUpdateLock().debug(out());
+				//TODO:check where that method is
```
Comment (adamretter) [inline extensions/commands/src/org/exist/commands/info/Lock.java]: As per previous comment

File: extensions/commands/src/org/exist/commands/info/Transactions.java [hunk 3]
```diff
@@ -47,7 +47,8 @@ public void process(XmldbURI collection, String[] commandData) throws CommandExc
 		try {
 			TransactionManager tm = BrokerPool.getInstance().getTransactionManager();
 			
-			tm.debug(out());
+			//TODO:check where that method is
+			//tm.debug(out());
```
Comment (adamretter) [inline extensions/commands/src/org/exist/commands/info/Transactions.java]: As per previous comment.

File: extensions/commands/src/org/exist/commands/threads/Deadlock.java [hunk 4]
```diff
@@ -21,11 +21,13 @@
...
 package org.exist.commands.threads;
 
+import java.util.Collections;
 import java.util.Map;
 
 import org.exist.plugin.command.AbstractCommand;
 import org.exist.plugin.command.CommandException;
-import org.exist.storage.lock.DeadlockDetection;
+//TODO:check where that class would be
+//import org.exist.storage.lock.DeadlockDetection;
```
Comment (adamretter) [inline extensions/commands/src/org/exist/commands/threads/Deadlock.java]: As per previous comment...

File: extensions/commands/src/org/exist/commands/threads/Deadlock.java [hunk 5]
```diff
@@ -45,7 +47,8 @@ public Deadlock() {
 	@Override
 	public void process(XmldbURI collection, String[] commandData) throws CommandException {
 		
-        Map<String, LockInfo> threads = DeadlockDetection.getWaitingThreads();
+		//TODO:check where that method is
+        Map<String, LockInfo> threads = Collections.emptyMap(); // DeadlockDetection.getWaitingThreads();
```
Comment (adamretter) [inline extensions/commands/src/org/exist/commands/threads/Deadlock.java]: As per previous comment...

File: extensions/debuggee/src/org/exist/debugger/client/DebugClient.java [hunk 6]
```diff
@@ -21,8 +21,9 @@
  */
 package org.exist.debugger.client;
 
-import jline.ConsoleReader;
-import jline.Terminal;
+//TODO:check where that class would be
+//import jline.ConsoleReader;
+//import jline.Terminal;
```
Comment (adamretter) [inline extensions/debuggee/src/org/exist/debugger/client/DebugClient.java]: As per previous comment...

File: extensions/debuggee/src/org/exist/debugger/client/DebugClient.java [hunk 7]
```diff
@@ -40,28 +41,29 @@
     private final static String COMMAND_STEP = "step";
     private final static String COMMAND_CONT = "cont";
 
-    private ConsoleReader console;
+    //TODO:check where that class would be
+    //private ConsoleReader console;
```
Comment (adamretter) [inline extensions/debuggee/src/org/exist/debugger/client/DebugClient.java]: As per previous comment...

File: extensions/debuggee/src/org/exist/debugger/client/DebugClient.java [hunk 8]
```diff
@@ -40,28 +41,29 @@
...
-                System.err.println("Exception caught: " + e.getMessage());
-            }
+//            try {
+//                String line = console.readLine(prompt + "> ");
+//                cont = parseInput(line);
+//            } catch (EOFException e) {
+//                break;
+//            } catch (IOException e) {
+//                System.err.println("Exception caught: " + e.getMessage());
+//            }
```
Comment (adamretter) [inline extensions/debuggee/src/org/exist/debugger/client/DebugClient.java]: As per previous comment...

**Comment by adamretter** [comment]
@reinhapa Is xmldb-api-1.7.0 the one with the optimistic read locking?

File: extensions/commands/src/org/exist/commands/info/Lock.java (hunk 1 above)
Comment (reinhapa) [reply extensions/commands/src/org/exist/commands/info/Lock.java]: The complete class does not compile due no such method exist on the `org.exist.collections.Collection`

File: extensions/commands/src/org/exist/commands/info/Lock.java (hunk 2 above)
Comment (reinhapa) [reply extensions/commands/src/org/exist/commands/info/Lock.java]: Ad previous comment :-)

File: extensions/commands/src/org/exist/commands/info/Transactions.java (hunk 3 above)
Comment (reinhapa) [reply extensions/commands/src/org/exist/commands/info/Transactions.java]: Ad previous comment :-)

File: extensions/commands/src/org/exist/commands/threads/Deadlock.java (hunk 4 above)
Comment (reinhapa) [reply extensions/commands/src/org/exist/commands/threads/Deadlock.java]: Imported class does not exist..

File: extensions/commands/src/org/exist/commands/threads/Deadlock.java (hunk 5 above)
Comment (reinhapa) [reply extensions/commands/src/org/exist/commands/threads/Deadlock.java]: Follow up problem of not existing import

File: extensions/debuggee/src/org/exist/debugger/client/DebugClient.java (hunk 6 above)
Comment (reinhapa) [reply extensions/debuggee/src/org/exist/debugger/client/DebugClient.java]: I did not find where those imported classes come from (trying to get eclipse to build all sources)

File: extensions/debuggee/src/org/exist/debugger/client/DebugClient.java (hunk 7 above)
Comment (reinhapa) [reply extensions/debuggee/src/org/exist/debugger/client/DebugClient.java]: Follow up problem based on the missing import

File: extensions/debuggee/src/org/exist/debugger/client/DebugClient.java (hunk 8 above)
Comment (reinhapa) [reply extensions/debuggee/src/org/exist/debugger/client/DebugClient.java]: Follow up problem based on the missing import

File: extensions/commands/src/org/exist/commands/info/Lock.java (hunk 1 above)
Comment (adamretter) [reply extensions/commands/src/org/exist/commands/info/Lock.java]: I see. Thanks

**Pull request closed without merging**